exporter/local/export.go

```diff
@@ -148,7 +169,74 @@ func (e *localExporterInstance) Export(ctx context.Context, inp *exporter.Source
 				return res
 			}
 
-			fs := fsutil.NewFS(src, walkOpt)
+			outputFS := fsutil.NewFS(src, walkOpt)
+
+			attestations, err = attestation.Unbundle(ctx, session.NewGroup(sessionID), inp.Refs, attestations)
+			if err != nil {
+				return err
+			}
+			if len(attestations) > 0 {
+				subjects := []intoto.Subject{}
+				err = outputFS.Walk(ctx, func(path string, info fs.FileInfo, err error) error {
+					if err != nil {
+						return err
+					}
+					if !info.Mode().IsRegular() {
+						return nil
+					}
+					f, err := outputFS.Open(path)
+					if err != nil {
+						return err
+					}
+					defer f.Close()
+					d := digest.Canonical.Digester()
+					if _, err := io.Copy(d.Hash(), f); err != nil {
+						return err
+					}
+					subjects = append(subjects, intoto.Subject{
+						Name:   path,
+						Digest: result.ToDigestMap(d.Digest()),
+					})
+					return nil
+				})
+				if err != nil {
+					return err
+				}
+
+				stmts, err := attestation.Generate(ctx, session.NewGroup(sessionID), inp.Refs, attestations, subjects)
+				if err != nil {
+					return err
+				}
+				stmtFS := staticfs.NewFS()
+
+				now := time.Now().UnixNano()
```

Comment:
Moved this out of the loop, but will keep the `Nano()` component for now - below we have `epoch.UnixNano()` so better to be consistent with that IMO.

Reply:
> below we have epoch.UnixNano() so better to be consistent with that IMO.

This is only because the fsutil field expects nanoseconds. The actual `SOURCE_DATE_EPOCH` var is in full seconds.